### DISCUSSION TITLE
Is the draft-7 hyper-schema-output $id correct?

### DISCUSSION THREAD

**Discussion by SorinGFS** [json-schema-org/community#330] [Q&A] [answered]
All the references in draft-07 are pointing to `http://json-schema.org/draft-07/...` while the `$id` of [hyper-schema-output](https://json-schema.org/draft-07/hyper-schema-output) is this: 

```json
{
    "$id": "http://json-schema.org/draft-7/hyper-schema-output",
    "...": "..."
}

Is that correct?

**Comment by jdesrosiers** [comment, marked as answer]
I'm sure that wasn't intentional. However, I'm not sure we should fix it. Someone somewhere might be relying on that typo.

**Reply by jdesrosiers** [reply]
Actually, the spec has the right URI, so we probably should fix it. https://datatracker.ietf.org/doc/html/draft-handrews-json-schema-hyperschema-01#section-4

**Reply by SorinGFS** [reply]
I also think it should be fixed, all references are pointing to the correct URL. I thought maybe it was intentionally set like that to disable the functionality (for whatever reason) although it would not be good to proceed like this. But if you say that wasn't intentional I will treat it like it was corrected.
It is curious that no one has noticed this for many years. It makes me think that this schema was not used at all.

Here is another one, this time from `vscode`:

```html
Problems loading reference 'https://json-schema.org/draft/2020-12/meta/hyper-schema': Unable to load schema from 'https://json-schema.org/draft/2020-12/meta/hyper-schema': Not Found. The requested location could not be found.
<!DOCTYPE html>
<html lang="en"><head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1"><!-- Begin.
```

I read something about this in `issues`, about some special behaviour of `$dynamicAnchor - $dynamicRef`, I'm not sure how exactly should those work in `https://json-schema.org/draft/2020-12/hyper-schema`, but `vscode` didn't understood either, and the file that is trying to download does not exist.

**Reply by jdesrosiers** [reply]
> It is curious that no one has noticed this for many years. It makes me think that this schema was not used at all.

I doubt it's seen much use. This schema just describes the output format, so there isn't much use for it. It's more of a reference. However, having a typo in the `$id` shouldn't make any difference. Nothing in that schema will evaluated differently due to the typo.

> Problems loading reference 'https://json-schema.org/draft/2020-12/meta/hyper-schema': Unable to load schema from 'https://json-schema.org/draft/2020-12/meta/hyper-schema': Not Found. The requested location could not be found.

It's true that vscode doesn't fully support 2020-12 (including dynamic references), but that looks like an error on our end hosting the vocabulary schema.

I'll look into correcting both the issue with typo in the draft-07 schema and the missing vocabulary schema. Thanks for letting us know!

**Reply by SorinGFS** [reply]
This [link](https://json-schema.org/draft/2020-12/meta/hyper-schema) still not working and I need that file, can you give me another one until you solve the problem?

**Reply by SorinGFS** [reply]
@jdesrosiers 

Now I see the problem where is comming from:

```json
{
    "$schema": "https://json-schema.org/draft/2020-12/hyper-schema",
    "$id": "https://json-schema.org/draft/2020-12/hyper-schema",
    "$vocabulary": {
        "https://json-schema.org/draft/2020-12/vocab/core": true,
        "https://json-schema.org/draft/2020-12/vocab/applicator": true,
        "https://json-schema.org/draft/2020-12/vocab/unevaluated": true,
        "https://json-schema.org/draft/2020-12/vocab/validation": true,
        "https://json-schema.org/draft/2020-12/vocab/meta-data": true,
        "https://json-schema.org/draft/2020-12/vocab/format-annotation": true,
        "https://json-schema.org/draft/2020-12/vocab/content": true,
        "https://json-schema.org/draft/2019-09/vocab/hyper-schema": true
    },
    "$dynamicAnchor": "meta",

    "title": "JSON Hyper-Schema",
    "allOf": [
        { "$ref": "https://json-schema.org/draft/2020-12/schema" },
        { "$ref": "https://json-schema.org/draft/2020-12/meta/hyper-schema" }
    ],
    "links": [
        {
            "rel": "self",
            "href": "{+%24id}"
        }
    ]
}
```

Probably was a decision to remove `meta/hyper-schema` file and its corresponding vocabulary was pointed to https://json-schema.org/draft/2019-09/vocab/hyper-schema but few rows bellow inexistent https://json-schema.org/draft/2020-12/meta/hyper-schema is rincluded.

What should I do as a quick fix to avoid the broken link error?

**Reply by jdesrosiers** [reply]
You can find the missing schema at https://github.com/json-schema-org/json-schema-spec/blob/2020-12/meta/hyper-schema.json until I get around to fixing it (hopefully tomorrow).

**Reply by SorinGFS** [reply]
Ok, thank you.

**Reply by jdesrosiers** [reply]
Both issues should be fixed now.

**Reply by SorinGFS** [reply]
Thank you very much!
